packages/nuxt/src/components/plugins/lazy-hydration-transform.ts

```diff
@@ -41,6 +54,20 @@ export const LazyHydrationTransformPlugin = (options: LoaderOptions) => createUn
       return isVue(id)
     },
     async transform (code) {
+      const codeWithoutComments = code
+        .replace(MULTI_LINE_COMMENT_RE, '')
+        .replace(SINGLE_LINE_COMMENT_RE, '\n')
+
+      const variables = new Set<string>()
+      if (codeWithoutComments !== code) {
+        for (const regex of EXCLUDE_RE) {
+          for (const match of codeWithoutComments.matchAll(regex)) {
+            if (!match[1]) { continue }
+            variables.add(match[1])
+          }
+        }
+      }
+
```

Comment:
I'm not sure this is quite right. Do we only want to scan variables if the code has comments? And I think we should respect 'scope' rather than looking for any variable anywhere in the code.

In general I'd prefer to avoid using regexp for this. I feel a parsing-based scope-tracking approach would be sounder (but we can make that performant by only parsing the code if it code contains some of the 'reserved' names - if not we have a 'fast path' without parsing)

Reply:
Thank you for pointing that out. I realized we probably shouldn’t restrict variable scanning to cases where comments are present, so I’ve submitted a corrected version accordingly.

By the way, just to clarify, were you suggesting that using a JavaScript AST parser such as `acorn` might be a more reliable approach compared to using regular expressions?